src/lib.rs

```diff
@@ -231,7 +233,14 @@ impl Adapter<HttpConnector, Body> {
     }
 
     async fn is_web_ready(&self, url: &Url, protocol: &Protocol) -> bool {
+        let start = Instant::now();
+        let interval = 5; // TODO: make this configurable?
```

Comment:
Let's make this 2 seconds. The default function timeout is 3 seconds. This will alert the users.